### PR TITLE
add iron fist ability

### PR DESCRIPTION
**Pull request by dyc3**

### CODE REVIEW

File: battle_abilities_test.go [hunk 1]
```diff
@@ -24,3 +24,24 @@ var _ = Describe("Scrappy", func() {
...
+			GeneratePokemon(PkmnBidoof, WithLevel(15), WithMoves(MoveCometPunch)), &a,
+		)
+		b.rng = AlwaysRNG()
+		Expect(b.Start()).To(Succeed())
+		t, _ := b.SimulateRound()
+		normalDamage := DamageDealt(t, target{1, 0})
+		boostedDamage := DamageDealt(t, target{0, 0})
+		Expect(normalDamage).To(BeNumerically(">", 0))
+		Expect(boostedDamage).To(BeNumerically(">", 0))
+		Expect(boostedDamage).To(BeNumerically(">", normalDamage))
```
Comment (adapap) [inline battle_abilities_test.go:43]: 2nd line here implies the first one is true.

File: battle_abilities_test.go [hunk 2]
```diff
@@ -24,3 +24,24 @@ var _ = Describe("Scrappy", func() {
...
+
+var _ = Describe("Ability: Iron Fist", func() {
+	a := Agent(new(dumbAgent))
+
+	It("should have punching moves deal 1.2x damage", func() {
+		b := New1v1Battle(
+			GeneratePokemon(PkmnMachop, WithLevel(15), WithAbility(AbilityIronFist), WithMoves(MoveCometPunch)), &a,
+			GeneratePokemon(PkmnBidoof, WithLevel(15), WithMoves(MoveCometPunch)), &a,
+		)
+		b.rng = AlwaysRNG()
```
Comment (adapap) [inline battle_abilities_test.go:36]: OOC, why not `SimpleRNG()`?

**Review by adapap — APPROVED** [review]
LGTM!

File: battle_abilities_test.go (hunk 2 above)
Comment (dyc3) [reply battle_abilities_test.go:36]: So the attacks don't miss

File: battle_abilities_test.go (hunk 1 above)
Comment (dyc3) [reply battle_abilities_test.go:43]: yeah but this gives a little bit more granularity for what went wrong

File: battle_abilities_test.go (hunk 2 above)
Comment (adapap) [reply battle_abilities_test.go:36]: The goal of simple RNG was to have a factory that produces attacks that always hit but don't have other side effects (stat changes, crits, etc.). Maybe it is too difficult now to trace this behavior?

**Pull request merged**